### PR TITLE
Add different confirmation message when attribute is used in a variation

### LINKED ISSUE #37162: [Enhancement]: Show a browser dialog when removing an attribute used for variations
When the user attempts to remove an attribute with the Used for variations box checked, we show a browser dialog window with an updated copy: `If you remove this attribute, customers will no longer be able to purchase some variations of this product.`

Screenshot 2023-03-09 at 23 14 23

**Designs:**
`zs56kcy53K7GoGhgyjcQ2R-fi-211%3A34103`

```[tasklist]
### Acceptance criteria
- [ ] Show a browser dialog when removing an attribute used for variations
```

### PR DESCRIPTION
**Pull request by nathanss**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add different confirmation message when deleting an attribute which is used in a variation

Closes #37162 .

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Create a new product in the legacy product manager
2. Select 'Variable' product type
3. Go to 'Attributes' tab and create an attribute with any name and values.
4. Go to 'Variations' tab and create a variation with the just created attribute.
5. Go back to 'Attributes' tab and remove the attribute.
6. Check that the confirmation message is 'If you remove this attribute, customers will no longer be able to purchase some variations of this product.'
7. Try to delete an attribute which doesn't have a variation
8. Check that the confirmation message is different.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js [hunk 1]
```diff
@@ -628,9 +628,19 @@ jQuery( function ( $ ) {
 		}
 	);
 
-	$( '.product_attributes' ).on( 'click', '.remove_row', function () {
-		if ( window.confirm( woocommerce_admin_meta_boxes.remove_attribute ) ) {
-			var $parent = $( this ).parent().parent();
+	$( '#product_attributes' ).on( 'click', '.product_attributes .remove_row', function () {
+		var confirmMessage = woocommerce_admin_meta_boxes.remove_attribute;
+		var $parent = $( this ).parent().parent();
+		if ( !$parent.is( '.taxonomy' ) ) {
```
Comment (octaedro) [inline plugins/woocommerce/client/legacy/js/admin/meta-boxes-product.js]: I think that this `if` will prevent showing the message when we delete a global attribute (even when it's actually being used for variations).
Maybe we should only check whether the 'Used for variations' checkbox is selected.

**Review by octaedro — COMMENTED** [review]
Good job @nathanss! The code looks mainly good, but when I try this with a global attribute, the message is not shown. I left a comment explaining that. Please ping me if you have any questions.

**Comment by nathanss** [comment]
@octaedro I updated the logic. Can you test again?

**Review by octaedro — APPROVED** [review]
LGTM! 🚀

**Pull request merged**